src/types/matrix.rs

```diff
@@ -270,19 +270,19 @@ impl $rust_name {
     /// This function multiplies the elements of the self matrix by the constant factor x. The
     /// result self(i,j) <- x self(i,j) is stored in self.
     #[doc(alias = $name _scale)]
-    pub fn scale(&mut self, x: f64) -> Value {
+    pub fn scale(&mut self, x: $rust_ty) -> Value {
         Value::from(unsafe { sys::[<$name _scale>](self.unwrap_unique(), x) })
     }
 
     /// This function adds the constant value x to the elements of the self matrix. The result
     /// self(i,j) <- self(i,j) + x is stored in self.
     #[doc(alias = $name _add_constant)]
-    pub fn add_constant(&mut self, x: f64) -> Value {
+    pub fn add_constant(&mut self, x: $rust_ty) -> Value {
         Value::from(unsafe { sys::[<$name _add_constant>](self.unwrap_unique(), x) })
     }
 
     #[doc(alias = $name _add_constant)]
-    pub fn add_diagonal(&mut self, x: f64) -> Value {
+    pub fn add_diagonal(&mut self, x: $rust_ty) -> Value {
         Value::from(unsafe { sys::[<$name _add_constant>](self.unwrap_unique(), x) })
```

Comment:
Just realized that `add_constant`  was incorrectly used! Sending a PR to fix that...

Reply:
merged your changes into `catompkins:multiroot_binding`